staticaddr/address/manager.go

```diff
@@ -111,7 +112,7 @@ func (m *Manager) NewAddress(ctx context.Context) (*btcutil.AddressTaproot,
 	m.Unlock()
 
 	// Ensure that we have that we have a sane current block height.
-	if m.currentHeight == 0 {
+	if m.currentHeight.Load() == 0 {
```

Comment:
This check could be removed if current height is correctly initialized.

Reply:
`Run` runs in background, so the caller doesn't know when `Run` has initialized `currentHeight`.
Additionally, it is better to double check even if it was initialized before, to detect a wrong order of methods called.